graphql/admin/http.go

```diff
@@ -175,14 +175,31 @@ func (gs *graphqlSubscription) Subscribe(
 		}
 	}
 
+	// Earlier the graphql-transport-ws library was ignoring the http headers in the request.
+	// The library was relying upon the information present in the request payload. This was
+	// blocker for the cloud team because the only control cloud has is over the HTTP headers.
+	// This fix ensures that we are setting the request headers if not provided in the payload.
+	httpHeaders, _ := ctx.Value("RequestHeader").(http.Header)
+	if len(httpHeaders) > 0 {
+		for k := range httpHeaders {
+			if len(strings.TrimSpace(reqHeader.Get(k))) == 0 {
+				reqHeader.Set(k, httpHeaders.Get(k))
+			}
+		}
+	}
+
 	req := &schema.Request{
 		OperationName: operationName,
 		Query:         document,
 		Variables:     variableValues,
 		Header:        reqHeader,
 	}
 	namespace := x.ExtractNamespaceHTTP(&http.Request{Header: reqHeader})
-	LazyLoadSchema(namespace) // first load the schema, then do anything else
+	glog.Infof("namespace: %d. Got GraphQL request over websocket.", namespace)
+	// first load the schema, then do anything else
+	if err = LazyLoadSchema(namespace); err != nil {
```

Comment:
:=

Reply:
Do you want to re-initialize the named return parameter `err`? I don't think that's idiomatic go. See a [function](https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/io/io.go;drc=bae7d772e800207ed764b70a9e5963f0c1e8f0aa;l=326-340) which I could find in the core lib which has similar named parameter. Inadvertently `:=` leads to variable shadowing which can cause issues I guess. I don't foresee what could fail in this case, but I remember reading that we should avoid variable shadowing with named return parameters.

Reply:
I didn't realize that. Now I think we should remove the named parameter completely. I don't think named parameters are a good idea.

Reply:
That sounds better. I have removed named parameters from the function. Apart from not needing named params here, we weren't using `payload` anywhere in the function. So it didn't really need a named param.